src/App.tsx

```diff
@@ -17,18 +48,36 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                changeStatus={setStatus}
+                changeQuery={setQuery}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {
+                isLoading ? (
+                  <Loader />
+                ) : (
+                  <TodoList
+                    todos={visibleTodos}
+                    changeTodo={setSelectedTodo}
+                    selectedTodo={selectedTodo}
+                  />
+                )
+              }
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal
+          selectedTodo={selectedTodo}
+          changeTodo={setSelectedTodo}
+        />
+      )}
```

Comment:
we render TodoModal only when selectedTodo exists

Reply:
i can't got it, explain please

Reply:
this comment related to the next one